File: memU-main/memU-main/memu/sdk/python/exceptions.py
"""
MemU SDK Exceptions

Custom exception classes for MemU SDK operations.
"""


class MemuSDKException(Exception):
    """Base exception for MemU SDK"""

    pass


class MemuAPIException(MemuSDKException):
    """Exception for API-related errors"""

    def __init__(
        self, message: str, status_code: int = None, response_data: dict = None
    ):
        super().__init__(message)
        self.status_code = status_code
        self.response_data = response_data


class MemuValidationException(MemuAPIException):
    """Exception for validation errors"""

    pass


class MemuAuthenticationException(MemuAPIException):
    """Exception for authentication errors"""

    pass


class MemuConnectionException(MemuSDKException):
    """Exception for connection errors"""

    pass

File: memU-main/memU-main/memu/sdk/javascript/src/exceptions.ts
/**
 * MemU SDK Exceptions
 *
 * Custom exception classes for MemU SDK operations.
 */

/**
 * Base exception for MemU SDK
 */
export class MemuSDKException extends Error {
  constructor(message: string) {
    super(message)

    this.name = 'MemuSDKException'

    // Maintains proper stack trace for where our error was thrown (only available on V8)
    if ('captureStackTrace' in Error)
      Error.captureStackTrace(this, MemuSDKException)
  }
}

/**
 * Exception for API-related errors
 */
export class MemuAPIException extends MemuSDKException {
  public responseData?: Record<string, unknown>
  public statusCode?: number

  constructor(message: string, statusCode?: number, responseData?: Record<string, unknown>) {
    super(message)

    this.name = 'MemuAPIException'

    if (statusCode !== undefined)
      this.statusCode = statusCode

    if (responseData !== undefined)
      this.responseData = responseData

    if ('captureStackTrace' in Error)
      Error.captureStackTrace(this, MemuAPIException)
  }
}

/**
 * Exception for authentication errors
 */
export class MemuAuthenticationException extends MemuAPIException {
  constructor(message: string, statusCode?: number, responseData?: Record<string, any>) {
    super(message, statusCode, responseData)

    this.name = 'MemuAuthenticationException'

    if ('captureStackTrace' in Error)
      Error.captureStackTrace(this, MemuAuthenticationException)
  }
}

/**
 * Exception for connection errors
 */
export class MemuConnectionException extends MemuSDKException {
  constructor(message: string) {
    super(message)

    this.name = 'MemuConnectionException'

    if ('captureStackTrace' in Error)
      Error.captureStackTrace(this, MemuConnectionException)
  }
}

/**
 * Exception for validation errors
 */
export class MemuValidationException extends MemuAPIException {
  constructor(message: string, statusCode?: number, responseData?: Record<string, any>) {
    super(message, statusCode, responseData)

    this.name = 'MemuValidationException'

    if ('captureStackTrace' in Error)
      Error.captureStackTrace(this, MemuValidationException)
  }
}
